checkov/terraform/checks/resource/aws/EC2PublicIP.py

```diff
@@ -0,0 +1,49 @@
+from checkov.common.models.enums import (
+    CheckCategories,
+    CheckResult,
+)
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class EC2PublicIP(BaseResourceCheck):
+
+    def __init__(self):
+        name = "EC2 instance should not have public IP."
+        id = "CKV_AWS_88"
+        categories = [CheckCategories.NETWORKING]
+        supported_resources = ['aws_instance', 'aws_launch_template']
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf):
+        """
+        Looks for if associate_public_ip_address is set
+            https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#associate_public_ip_address
+            or
+            https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#associate_public_ip_address
+
+        :param conf: dict of supported resource configuration
+        :return: <CheckResult>
+        """
+        # For aws_instance
+        if 'associate_public_ip_address' in conf.keys():
+            if conf['associate_public_ip_address'] == [True]:
```

Comment:
I'm assuming the reviewer might know Terraform better than I do, what does doing `= "true"` instead of `true` for things Terraform considers should be booleans, do?

Reply:
Err, okay then https://medium.com/@xtg/terraform-boolean-evaluation-b866b528e90b, I guess my code is okay.